### ISSUE TITLE
Firefox occaisionally doesn't render pairs of videos on initial join

### ISSUE THREAD

**Issue by hensmi-amazon** [aws/amazon-chime-sdk-js#2078]
### What happened and what did you expect to happen?

Making an issue for discussion on https://github.com/aws/amazon-chime-sdk-js/pull/1831 just so it doesn't get lost. I was able to reproduce (maybe a similar issue) an unsignalled SSRC related issue in FF with network manipulation and pushed a backend change to not negotiate MID on client receive streams, which fixed the issue for me, however it still occasionally reproduces. It appears to resolve (with same possibility for issue) on resubscribes, however we have been reducing unnecessary resubscribes as part of unrelated changes.

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

* Start 2 videos in call
* Join from FF and observe videos 

### Amazon Chime SDK for JavaScript version

2.27

### What browsers are you seeing the problem on?

FF

### Browser version

Latest

### Browser console logs

See linked PR

**Comment by hensmi-amazon** [comment]
@haofiltered Looking at last logs, I was just going to say that the backend changes were deployed (I can tell they were in those logs by the SDP), I wonder if this is some other FF related issue then the one fixed.
* Whats the repro rate? It was only like 1/10 for me without manual network manipulation, in which case I was able to get 100% (but that was fixed by backend changes). 
* Does changing region impact this repro rate?
* Does toggling video on local or remote help recover? 

If easy to reproduce (because I can no longer repro unfortunately) can you try with FF debug logs? Something like the following
```
export LOG_PATH=$HOME/etc/debug // Or where desired

export MOZ_LOG=MediaManager:5,GMP:5,GetUserMedia:5,CamerasParent:5,CamerasChild:5,MediaStream:5,MediaStreamGraph:5,MediaRecorder:5,MediaEncoder:5,TrackEncoder:5,webrtc_trace:5,signaling:5,mtransport:5,PlatformDecoderModule:5
export MOZ_LOG_FILE=$LOG_PATH/moz.log

# Signaling logs (SDP).  Note these commands should be unnecessary, it should
# all go to moz log but they are just here anyways for future debugging
export NSPR_LOG_FILE=$LOG_PATH/nspr.log
export NSPR_LOG_MODULES=signaling:5,webrtc_trace:5,mtransport:5,timestamp:1

# ICE/STUN/TURN logs:
export R_LOG_LEVEL=9
export R_LOG_DESTINATION=stderr # @hensmi has been unable to get this to work with a file path
export R_LOG_VERBOSE=1 # include the module name generating the message
export WEBRTC_TRACE_FILE=nspr

/Applications/Firefox.app/Contents/MacOS/firefox > $LOG_PATH/ice.log 2>&1
```

**Comment by haofiltered** [comment]
Hi @hensmi-amazon 

- I can reproduce it almost every time. Just using FF to join the meeting with camera off by default, don't call `startLocalVideoTile()`.
- Didn't test the region, but I think it has no impact.
- Toggle video on remote does help with recovery.

Not sure how to get the FF debug logs, can you be more specific? Here is the log by setting `LogLevel.DEBUG`.[logs2.txt](https://github.com/aws/amazon-chime-sdk-js/files/8163875/logs2.txt)
And you can see that remote user toggle video did solve the issue from this log.

**Comment by hensmi-amazon** [comment]
I will take a look again now, had to come back to this. For the debug logs (assuming you are on MacOS) you can copy that into a shell script (e.g. `firefox_debug.sh`, then quit firefox, and then run it to restart.
```
mkdir -p $HOME/etc/debug
chmod +x firefox_debug.sh
./firefox_debug.sh
```
Then can reproduce and attach a tarball of the logs to this ticket.

I'm hoping I can find a mitigation that doesn't require spamming SDP negotiations until all videos are received....which is sorta what we were accidentally doing until https://github.com/aws/amazon-chime-sdk-js/pull/1937

**Comment by haofiltered** [comment]
Here is the debug log, thank you @hensmi-amazon
I assume you only need `ice.log`, but I compressed the whole folder just in case.
Let me know if you need anything else. I am using latest `2.28.0` for this debug log.
[debug.zip](https://github.com/aws/amazon-chime-sdk-js/files/8209474/debug.zip)

**Comment by hensmi-amazon** [comment]
@haofiltered do you have meeting ID? its unfortunatly not in debug logs. Thanks for those, I can see a similar issue around unsignalled SSRCs (though I am not having luck reproducing myself :( )
```
[Child 2724: WebrtcCallThread #1]: I/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1522: VideoConduit 12871f000: Switching remote SSRC from 0 to 292709940
[Child 2724: WebrtcCallThread #1]: D/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:993: SetRemoteSSRCConfig: SSRC 292709940 (0x11726634)
[Child 2724: WebrtcCallThread #1]: V/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1532: VideoConduit 12871f000: Received RTP packet, seq# 57377, len 1178, SSRC 292709940 (0x11726634) 
[Child 2724: WebrtcCallThread #1]: D/webrtc_trace (call.cc:1351): receive_rtp_config_ lookup failed for ssrc 292709940
[Child 2724: WebrtcCallThread #1]: E/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1494: DeliverPacket DeliverPacket Failed for RTP packet, 1
[Child 2724: WebrtcCallThread #1]: D/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:983: NotifyUnsetCurrentRemoteSSRC (12871f000): Unsetting SSRC 292709940 in other conduits
...
[Child 2724: WebrtcCallThread #1]: I/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1522: VideoConduit 12871f000: Switching remote SSRC from 292709940 to 3228089821
[Child 2724: WebrtcCallThread #1]: D/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:993: SetRemoteSSRCConfig: SSRC 3228089821 (0xc068bddd)
[Child 2724: WebrtcCallThread #1]: V/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1532: VideoConduit 12871f000: Received RTP packet, seq# 37699, len 1137, SSRC 3228089821 (0xc068bddd) 
[Child 2724: WebrtcCallThread #1]: D/webrtc_trace (call.cc:1351): receive_rtp_config_ lookup failed for ssrc 3228089821
[Child 2724: WebrtcCallThread #1]: E/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1494: DeliverPacket DeliverPacket Failed for RTP packet, 1
[Child 2724: WebrtcCallThread #1]: D/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:983: NotifyUnsetCurrentRemoteSSRC (12871f000): Unsetting SSRC 3228089821 in other conduits
[Child 2724: WebrtcWorker #2]: D/webrtc_trace (probe_controller.cc:381): kWaitingForProbingResult: timeout
[Child 2724: Socket Thread]: D/webrtc_trace (rtp_utility.cc:360): Failed to find extension id: 4
[Child 2724: WebrtcCallThread #1]: V/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1532: VideoConduit 12871f000: Received RTP packet, seq# 37700, len 1137, SSRC 3228089821 (0xc068bddd) 
[Child 2724: WebrtcCallThread #1]: D/webrtc_trace (call.cc:1351): receive_rtp_config_ lookup failed for ssrc 3228089821
[Child 2724: WebrtcCallThread #1]: E/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1494: DeliverPacket DeliverPacket Failed for RTP packet, 1
[Child 2724: WebrtcCallThread #1]: V/signaling [WebrtcCallThread #1|WebrtcVideoSessionConduit] VideoConduit.cpp:1532: VideoConduit 12871f000: Received RTP packet, seq# 37701, len 1137, SSRC 3228089821 (0xc068bddd) 
[Child 2724: WebrtcCallThread #1]: D/webrtc_trace (call.cc:1351): receive_rtp_config_ lookup failed for ssrc 3228089821
```

Looks like its thrashing between two SSRCs (why it needs 2+ videos to repro) before receiving the SDP and can't recover after. Will continue looking.

**Comment by haofiltered** [comment]
@hensmi-amazon 
Couldn't find the meetingID, so I just generated another debug log with meetingID down below:
MeetingID: `83b56da8-a7c9-422a-9dd0-3df276860706`
[debug.zip](https://github.com/aws/amazon-chime-sdk-js/files/8217694/debug.zip)

No need 2+ videos to reproduce, I reproduced this time with only one remote video.

So here is what I did:
1. Join a user (A) using Chrome, with video on.
2. Join another user (B) with Firefox, and **default video off**.   *Make sure you don't call `audioVideo.startLocalVideoTile()` for this user. You should be able to repro. 

For any user joined before user B joined, there is a chance that their video tile will be black on user B's screen. For any new users joined after user B joined, there will be no issue, their video tile will show up normally on user B's screen.

**Comment by hensmi-amazon** [comment]
So I was able to reproduce, its actually same issue as before, reintroduced with an adjustment for chrome simulcast...browsers are fun. Basically (perhaps not information interesting to everyone...):

* FF has  unsignalled SSRC bug that breaks video seemingly 100% of the time RTP comes before SDP with how we negotiated SDP. Previously we unintentionally mitigated this without knowing before, by spamming renegotiation (often concurrently) at beginning of call. Resubscribe spam was reduced as part of the following critical simulcast fix: https://github.com/aws/amazon-chime-sdk-js/pull/1937
* I could mitigate this (covered by earlier discussion) by not putting MID in client-receive media sections. Transmitting MID didn't seem to help. Also by chance did for inactive sections.
* Chrome simulcast will drop all but one simulcast layer if you ever remove MID (needs for stream determination i guess), even in inactive sections, and won’t re-enable if added again resulting on issues in toggle: https://github.com/aws/amazon-chime-sdk-js/issues/2064. So I added to inactive sections, and just remove from client-receive sections.
* The FF mitigate breaks if any inactive sections have MID (note: *not* when it is in send section….and even if the inactive section is only ever switched to send section). So I had to again adjust it, it now negotiates it only on sections without RIDs. I may need to re-evaluate this if/when we add simulcast support to FF.

Tldr: Need to roll out another server side change. This will complete mid next week and i will update this ticket then.

**Comment by haofiltered** [comment]
Thank you @hensmi-amazon !
Glad you could reproduce on your end now. Looking forward to the fix.
And just FYI, I had `simulcast` **off** while reproducing this issue, just trying to provide as more info as possible, hopefully can help you with the fix a little bit. :)

**Comment by hensmi-amazon** [comment]
Backend change has been deployed! I retested with a 200 ms delay on the signal channel which previously would result in 100% repro and it rendered fine. I also made sure i didn't break simulcast again. Will close this for now, thats for help!

**Comment by haofiltered** [comment]
Thank you very much for helping! @hensmi-amazon
